source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -837,6 +837,11 @@ Envoy::Ssl::CertificateDetailsPtr ContextImpl::certificateDetails(X509* cert,
         *certificate_details->add_subject_alt_names();
     subject_alt_name.set_uri(uri_san);
   }
+  for (auto& ip_san : Utility::getSubjectAltNames(*cert, GEN_IPADD)) {
```

Comment:
Can you add a test?

Reply:
As I mentioned in the PR description, it is difficult to add a test for this because we do not have IP SAN tests in our test certs and I could not find any easy way to mock this stuff (not sure if it is possible). 
Please let me know if you have any ideas, I will try.

Reply:
@ggreenway @snowp @lizan any thoughts here on how to test this? I feel like we should be able to do it somehow?

Reply:
```
...
[v3_req]
subjectAltNames = @alt_names
...

[alt_names]
IP.1 = 1.1.1.1
```
in the .cfg file would generate a cert with the ip SAN. Look at san_dns_cert.cfg for an example of a cert configuration with SANs

Reply:
@snowp Thank you. Let me try that. What do I have to run after adding that SAN?

Reply:
test/extensions/transport_sockets/tls_test_data/certs.sh

you probably want to comment out the commands to generate certs you didnt touch to avoid blowing up the diff

Reply:
ok. Cool. Thanks. Will try